diff --git a/lib/index.cjs b/lib/index.cjs
index 746821be4feef9b2ede2300b91ed23e99154ac84..d76c450e554f990482a9a675301116be4d9ebea4 100644
--- a/lib/index.cjs
+++ b/lib/index.cjs
@@ -2329,7 +2329,11 @@ var readBrowserText = new Function(
                 const text = a.textContent?.replace(/\s+/g, ' ').trim()
                 const href = a.getAttribute('href')
                 return text && href
-                    ? \`- [\${text}](\${new URL(href, location.href).href})\`
+                    ? '- [' +
+                          text +
+                          '](' +
+                          new URL(href, location.href).href +
+                          ')'
                     : ''
             })
             .filter(Boolean)
diff --git a/lib/index.mjs b/lib/index.mjs
index a5ebddcd2ebd4163fa064f2f202adf8b3027233f..b9717ac132f559d36b2e6dda8e6c5f3142f10171 100644
--- a/lib/index.mjs
+++ b/lib/index.mjs
@@ -2304,7 +2304,11 @@ var readBrowserText = new Function(
                 const text = a.textContent?.replace(/\s+/g, ' ').trim()
                 const href = a.getAttribute('href')
                 return text && href
-                    ? \`- [\${text}](\${new URL(href, location.href).href})\`
+                    ? '- [' +
+                          text +
+                          '](' +
+                          new URL(href, location.href).href +
+                          ')'
                     : ''
             })
             .filter(Boolean)
